api/proto/config/edgeview.proto

```diff
@@ -0,0 +1,50 @@
+// Copyright(c) 2021 Zededa, Inc.
+// All rights reserved.
+
+syntax = "proto3";
+
+package org.lfedge.eve.config;
+option go_package  = "github.com/lf-edge/eve/api/go/config";
+option java_package = "org.lfedge.eve.config";
+
+message EdgeViewConfig {
+    // JWT token for signed info, it contains the dispatcher
+    // endpoint IP:Port, device UUID and expiration time
+    string  token = 1;
+    // dispatcher certificate(s) if it's not well-known CA signed
+    repeated bytes disp_cert_pem = 2;
+    // policy for device access through edge-view
+    DevEdgeViewPolicy dev_policy = 3;
+    // policy for apps through edge-view
+    AppEdgeViewPolicy app_policy = 4;
+}
+
+message DevEdgeViewPolicy {
+    // device side of edge-view access is alllowed or not
+    bool allow_dev = 1;
+    // allow ssh into device or not, ssh key still need to be
+    // installed separately even the edge-view allows it
+    bool ssh = 2;
+    // allow to issue reboot device or not
+    bool reboot = 3;
+    // allow to launch VNC into application console or not; the
+    // app still need to enable VNC access even it is allowed here
+    bool app_console = 4;
+}
+
+message AppEdgeViewPolicy {
+    // app side of edge-view access is allowed or not
+    bool allow_app = 1;
+    // ssh through port 22 into app is allowed or not
+    bool ssh = 2;
+    // the https proxy is allowed or not
+    bool proxy = 3;
+    // only the defined set of tcp ports allowed
+    repeated int32 only_ports = 4;
+    // the ports listed are not allowed
+    repeated int32 disallow_ports = 5;
+    // only the listed IP-Addresses or Domain Names are allowed
+    repeated string only_addrs = 6;
+    // the IP-Addresses or Domain Names are not allowed
+    repeated string disallow_addrs = 7;
```

Comment:
How do you see the relationship between this and the current ACLs?

Would it make sense to instead have settings on the controller which disallow/restrict the set of ACLs which can be set i.e., EVE-OS would just see the same ACLs and the restrictions for portmap and outbound IP, domainnames, and ports are enforced in the controller against the ACLs?

Reply:
I currently do not see the relation from the current ACLs.
  1) this policy intention is only from the edge-view point-of-view
  2) current ACLs are app-bundle based, at the time of app-bundle there is no ip address knowledge.
     and also the ip addresses are dynamic, can change from reboot-to-reboot
  3) if people don't want ACLs since they expose to everyone on the network or even Internet, but edge-view can
     still access those ports, thus may need restrictions
  4) it's probably quite hard to tie this together, possibly would be quite buggy
  5) but it's controller's code later on, if we can figure out a good way, this api should be able to handle ok

 - Is the intent that this cmd_option in the info messages can be used to see all of the commands which are excuted?
That wouldn't work since we assume that if we have a failure to send an info message then when we retry sending that info message we will include the most recent information.
This if cmd_option was xyzzy and that failire to send, then the operation finished and we attempt send an info with cmd_option being empty, the controller would never see that xyzzy was performed.

Reply:
Just because the implementation has this enforcement in different places doesn't mean it makes any sense to the users; they have no idea how our implementation is structured.

Presumably a user would want to know whether port X of the app instance can be access, and if so if there are restrictions on who can access it. Thus it would make sense to have model to describe this in the EVE API which captures that. Duplicating the port numbers in two places definitely doesn't help.

Reply:
Does it make sense just to remove 'ports', 'addrs' for now, only use the 'allow_app', 'allow_dev' those boolean for now to get this started. The proper/detail policy design can be figured out later?

Reply:
Sorry for not getting back to you earlier with the break etc.

I think it is fine to start with a course-grained allow_app and allow_dev setting (plus the JWT) to limit access.

If we want more fine-grained access control to the app instances it almost seems like this is an additional pseudo-network interface with ACLs. By that I mean that an app instance might have an indirect interface (e.g. attached to a local network instance) with one set of ACLs, and a direct interface (e.g., attached to a switch network instance) with a different set of ACLs, and potentially we want to describe the allowed edge-view debug access as having a third set of ACLs (and resulting in traffic flowing to the app instance over either its indirect or direct interface).
If we think that makes sense then down the road we can add an edge-view ACL message in the API to specify this. Do you think that makes sense?

Reply:
Yes, I can right now only leave allow_app and allow_dev (JWT is automatic) there.
For the ACLs, it's a little different from our current download ACEs, in which this needs a destination ip address and port, rate probably does not apply here. I was only check those during the tcp session setup time, but if we want to check like normal ACL, we may need to install them into the kernel on the 'bnx' bridges; also i'm not sure how to do this when it's a switch-NI case, the packets will go out through the mgmt intf to reach the app ip address in that case. We can leave this app ACL thing to later I guess.